pkgs/development/mobile/androidenv/test-suite.nix

```diff
@@ -1,9 +1,11 @@
-{ stdenv, callPackage }:
+{}:
 let
-  examples-shell = callPackage ./examples/shell.nix {};
-  examples-shell-with-emulator = callPackage ./examples/shell-with-emulator.nix {};
+  pkgs = import ../../../.. {};
+
+  examples-shell = pkgs.callPackage ./examples/shell.nix {};
+  examples-shell-with-emulator = pkgs.callPackage ./examples/shell-with-emulator.nix {};
 in
-stdenv.mkDerivation {
+pkgs.stdenv.mkDerivation {
```

Comment:
```
$ nix build ".#androidenv.test-suite" --impure
this derivation will be built:
  /nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv
building '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv'...
androidenv-test-suite> unpacking sources
androidenv-test-suite> unpacking source archive /nix/store/yjcmw1b3637kxnb0sy0ysidhga71h2ad-androidenv
androidenv-test-suite> source root is androidenv
androidenv-test-suite> patching sources
androidenv-test-suite> installing
androidenv-test-suite> no Makefile or custom installPhase, doing nothing
androidenv-test-suite> post-installation fixup
error: builder for '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv' failed to produce output path for output 'out' at'/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv.chroot/nix/store/0p420m7lrma1xi4p9ja04r8s2v89xifi-androidenv-test-suite'
```

Reply:
Based on the README file, I'm running it like this
```
nix-build ./default.nix -A androidenv.test-suite.passthru.tests
```
and Ofborg could run them successfully before removing the `alowUnfree = true`! Also I'm getting
```
nix build ".#androidenv.test-suite" --impure --extra-experimental-features nix-command --extra-experimental-features flakes         
error: builder for '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv' failed to produce output path for output 'out' at '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv.chroot/nix/store/0p420m7lrma1xi4p9ja04r8s2v89xifi-androidenv-test-suite'
```
which I have no idea why it failed on your side, but it gives me something else here! Can we stay with the stable versions and fix this problem whenever `flakes` get stable?

Reply:
```
$ nix-build -A androidenv.test-suite
this derivation will be built:
  /nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv
building '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv'...
unpacking sources
unpacking source archive /nix/store/yjcmw1b3637kxnb0sy0ysidhga71h2ad-androidenv
source root is androidenv
patching sources
installing
no Makefile or custom installPhase, doing nothing
post-installation fixup
error: builder for '/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv' failed to produce output path for output 'out' at'/nix/store/2bv22ji2jlcp7py5v6m0r0vg03nn3v14-androidenv-test-suite.drv.chroot/nix/store/0p420m7lrma1xi4p9ja04r8s2v89xifi-androidenv-test-suite'
```

the androidenv-test-suite derivation doesn't produce any outputs, you probably want to use the 2 tests in the derivation so they're the dependencies on androidenv-test-suite and run `touch $out` in some phase

Reply:
@Artturin thank you so much. With your help it's working now. check it in [70fcbcd](https://github.com/NixOS/nixpkgs/pull/213871/commits/70fcbcdb2573d0abe95c98fdcbec8cbfa774eabb). Let me know if I need to change anything.